src/main/java/ch/njol/skript/expressions/ExprElement.java

```diff
@@ -60,28 +66,38 @@ public class ExprElement extends SimpleExpression<Object> {
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		expr = LiteralUtils.defendExpression(exprs[1]);
 		number = (Expression<Number>) exprs[0];
-		element = parseResult.mark;
+		type = ElementType.values()[parseResult.mark];
 		return LiteralUtils.canInitSafely(expr);
 	}
 
 	@Override
 	@Nullable
 	protected Object[] get(Event e) {
-		Object[] os = expr.getArray(e);
-		if (os.length == 0)
+		Iterator<?> iter = expr.iterator(e);
+		if (iter == null || !iter.hasNext())
 			return null;
-		Object o;
-		if (element == -1) {
-			o = os[0];
-		} else if (element == 1) {
-			o = os[os.length - 1];
-		} else if (element == 2) {
-			Number number = this.number.getSingle(e);
-			if (number == null || number.intValue() - 1 >= os.length || number.intValue() - 1 < 0)
-				return null;
-			o = os[number.intValue() - 1];
-		} else {
-			o = CollectionUtils.getRandom(os);
+		Object o = null;
```

Comment:
IMO it would be better to give it a clearer name like `obj` at least - optional
```suggestion
		Object obj = null;
```